docs/README.md

```diff
@@ -19,417 +17,498 @@
 
-### Pull Requests
+---
 
-Validate pull requests for mergeability based on content and structure of your PR (title, labels, milestone, project, description, approvals, etc). The rule-sets are fully configurable. Here are a few examples of what you can do:
 
-- Prevent accidental merging of Pull Requests that are work in progress by labeling it `wip` or prefixing the title with the abbreviation.
+# Usage
 
-- Ensure all Pull Requests have a description so that when you view through history you still have context.
+1. [Install](https://github.com/apps/mergeable) the Mergeable GitHub App.
+2. [Create](#configuration) your recipe(s). Here are some [examples](#examples).
+3. Commit and push the recipes to your repository at `.github/mergeable.yml`
 
-- Ensure that all Pull Requests are approved by a specific list of users. It is especially useful if one of the users is not a collaborator of your repository -- something GitHub does not already support.
+> ☝ **NOTE:** You can also [deploy to your own server](deploy.md).
 
-- Ensure that all Pull Requests merged are in a specific GitHub Project. Mergeable even detects when you are closing an issue that is associated with the right project. This is very useful when your process includes QA validation of stories.
+# Configuration
 
-- [And more](#configuration).
+**Mergeable** is **highly** configurable.
+Define your recipes by creating a `.github/mergeable.yml` file in your repository.
 
-### Issues
+## Basics
+The configuration consists of any number of recipes. Recipes are created by tying [events](#events) with a set of [validators](#validators) and [actions](#actions) together:
 
-There may be certain formats and structure you want your Github issues to adhere to. Mergeable allows you to configure this and will create a comment with a list of suggested improvements to your Issue.
+```yml
+version: 2
+mergeable:
+  - when: {{event}}, {{event}} # can be one or more
+    validate:
+      # list of validators. Specify one or more.
+      - do: {{validator}}
+        {{option}}: # name of an option supported by the validator.
+          regex: {{expression}} # where regex is a nested option
+    pass: # list of actions. Specify one or more. Omit this tag if no actions are needed.
+      - do: {{action}}   
+    fail: # list of actions. Specify one or more. Omit this tag if no actions are needed.
+      - do: {{action}}            
+    error: # list of actions. Specify one or more. Omit this tag if no actions are needed.
+      - do: {{action}}      
+```
 
-- Notify the author and collaborators when an issues does not adhere to certain formats in the title.
+Take a look at some [example recipes](#examples).
 
-- Notify when projects and milestones are not associated.
+> ☝ **NOTE:** Earlier versions used a [different set of convention](version1.md#configuration). Don't worry, It is fully compatible.
```

Comment:
Similar to my comment in the v1 config doc, i think we should state that it may be not be fully support in the future versions so we are not always obligated to keep that config forward, which maybe difficult if the system gets more and more complex.

Reply:
I updated the verbage. LMK what you think.